ow_lander/scripts/mechanical_power_arm.py

```diff
@@ -0,0 +1,65 @@
+#!/usr/bin/env python
+
+import rospy
+from sensor_msgs.msg import JointState
+from std_msgs.msg import Float64
+
+# Python libs
+import sys, time
+import numpy as np
+
+#history = []
+
+VERBOSE=True
+
+class MechanicalPower:
+    def __init__(self):
+        '''Initialize ros publisher, ros subscriber'''
+        # topic where we publish
+        self.power_pub = rospy.Publisher('/mechanical_power/raw', Float64, queue_size = 10)
+        self.power_pub_a = rospy.Publisher('/mechanical_power/average', Float64, queue_size = 10)
+        
+        self.subscriber = rospy.Subscriber('/joint_states', JointState, self.callback, queue_size = 1)
+        self.history = []
+        rate = rospy.Rate(100)    
+            
+
+
+
+    def _moving_average(self, msg):
+        #global history
+        self.history.append(msg)
+        if len(self.history) > 10:
+            self.history = self.history[-10:]
+            average = sum(self.history) / float(len(self.history))
+            #rospy.loginfo('Average of most recent {} samples: {}'.format(len(self.history), average))
+        else:
+            average = msg
+        return average    
+        
+    def callback(self, ros_data):
+        self._value = len(ros_data.name)
+        l = len(ros_data.name)
```

Comment:
"l" isn't much of a variable name. How about "num_joints" or something descriptive?

Reply:
Done!